tasks/main.yml

```diff
@@ -23,6 +23,17 @@
     state: present
   when:
     - not network_packages is subset(ansible_facts.packages.keys())
+  register: package_install
+
+# If network packages have changed and wireless connections are required,
+# NetworkManager must be restarted
+- name: Restart NetworkManager
```

Comment:
Yeah I am not too happy with this implementation, but restarting NM prior to setting up the connections was the only way I could get wireless to work if NetworkManager-wifi was not previously installed.

I would say it's likely that the main use case for using this role to set up wireless connections would be running the role locally (e.g. on a laptop). I think in this case it would be safe to restart NM.

Reply:
@pcahyna @richm should we fail here or restart NM?

Reply:
Consider adding a variable `network_nm_restart_ok` - if `true`, this means it is OK for the role to restart nm as needed.  If `false`, this means the role should emit some sort of message like "These changes require NetworkManager restart to take effect.  Please restart it when it is safe to do so."  I suppose the defaults/main.yml setting should be `network_nm_restart_ok: false`

Reply:
Using such a variable sounds fine by me. It would like to drop the `nm` infix to make it provider agnostic. How about

`network_allow_restart`?

@thom311 what is the impact of restarting NM? What is your opinion? I guess the biggest problem might be breaking VPN connections but we do not support them, yet.

Reply:
I have added an network_allow_restart variable that defaults to 'yes'. I think it would be unlikely that a user would not want NetworkManager restarted if they are configuring wireless connections but we can at least give them the option to turn it off. Probably the most common use case for this would be configuring a laptop or workstation device locally where the connection dropping out is not a huge deal

Please let me know what you think

Reply:
> Using such a variable sounds fine by me. It would like to drop the `nm` infix to make it provider agnostic. How about
> 
> `network_allow_restart`?
> 
> @thom311 what is the impact of restarting NM? What is your opinion? I guess the biggest problem might be breaking VPN connections but we do not support them, yet.

@thom311 please take a look here.

> I have added an network_allow_restart variable that defaults to 'yes'. I think it would be unlikely that a user would not want NetworkManager restarted if they are configuring wireless connections but we can at least give them the option to turn it off. Probably the most common use case for this would be configuring a laptop or workstation device locally where the connection dropping out is not a huge deal
> 
> Please let me know what you think

The other perspective is that laptops or workstations will probably already have NetworkManager-wifi installed, so they will not benefit from this and defaulting to `false` for `network_allow_restart` should not be an issue for most users but still be the safer default.

Reply:
Could the role determine whether it is safe to restart? If the only problem expected is with VPNs, the role could detect whether a VPN is being used and use `false` as the default value for `network_allow_restart`, otherwise use `true`.

Reply:
NetworkManager might also forget other runtime information when being restarted, so the it would be best if users restarted the system for deterministic behavior.

Also, just for creating the profiles (persistent_state:present, no state), the wifi plugin would not be needed. It is only needed to bring the profiles up. So technically, this should be checked here, as well. I am currently more inclined not to restart NM in the role and request the user to do this. This would also simplify testing. What do you think?

Reply:
@jaedolph would it work for you if we default restarting to false and re-consider if there is use feedback that tells us to enable it? I am just wondering how we could unblock this for now since we do not have any real user stories about how/when this would be used.

Reply:
I think this is a good idea, probably safest to default to not restarting. I will have a look at this

Reply:
I changed the default to "no" so we can merge it.

Reply:
Thanks @tyll, sorry that I was so slow at getting to this